src/Simplify_Exprs.cpp

```diff
@@ -53,6 +53,36 @@ Expr Simplify::visit(const Broadcast *op, ExprInfo *bounds) {
 
 Expr Simplify::visit(const VectorReduce *op, ExprInfo *bounds) {
     Expr value = mutate(op->value, bounds);
+    if (const Shuffle *shuffle = value.as<Shuffle>()) {
+        int broadcast_factor = shuffle->is_broadcast();
+        if (broadcast_factor > 1) {
+            // How many lanes does the smaller vector reduce have?
+            int reduced_result_lanes = std::max(1, op->type.lanes() / broadcast_factor);
+
+            Expr reduced = VectorReduce::make(op->op, shuffle->vectors[0], reduced_result_lanes);
+            if (reduced_result_lanes < op->type.lanes()) {
+                reduced = Shuffle::make_broadcast(reduced, op->type.lanes() / reduced_result_lanes);
+            }
+            if (reduced.type().lanes() != op->type.lanes()) {
```

Comment:
When can these things be equal and not one? Does this handle vector reductions that aren't total?